### ISSUE TITLE
Question about 'Predict' syntax and differences in predicted outputs

### ISSUE THREAD

**Issue by smithja16** [hmsc-r/HMSC#171]
Hi again. This is similar to the question below about 'prepareGradient' - sorry for starting a new thread, but my question is a little different.

I'm hoping for assistance identifying 1) why the following two methods for predicting new data give me different results, and 2) which is the result I want to use.

'my_model' is fitted with spatial random effects (lat lon coordinates, GPP method with Knots); fitting data has 1387 rows
'grid_data' is new X data at a grid of new xy coordinates; data has 365 rows

xycoords = as.matrix(grid_data[,c("longitude", "latitude")])
colnames(xycoords) <- c("x-coordinate","y-coordinate")
rownames(xycoords) <- 1:nrow(xycoords)

**Method 1:**
gradient <- prepareGradient(my_model, XDataNew=grid_data, sDataNew=list(sample=xycoords))
#I get a warning here: In sprintf("new_unit", 1:(nyNew)) : one argument not used by format 'new_unit'
predY1 <- Hmsc:::predict.Hmsc(my_model, XData=grid_data, Gradient=gradient, expected=T, predictEtaMean=T)
#specifying predictEtaMean T or F makes similar maps

predY1_mean <- apply(abind(predY1,along=3), c(1,2), mean)
#then plot as raster

**Method 2:**
studyDesignNew = data.frame(sample=as.factor(1:nrow(grid_data)), vessel=as.factor(grid_data$vessel_ID))  #same structure as fitted model
rL_v <- HmscRandomLevel(units=studyDesignNew$vessel) 
rL_spat <- HmscRandomLevel(sData = xycoords)
rLs <- list(sample=rL_spat, vessel=rL_v)
predY2 <- Hmsc:::predict.Hmsc(hm_pres, XData=grid_data, studyDesign=studyDesignNew, ranLevels=rLs, expected=T, predictEtaMean=T)

predY2_mean <- apply(abind(predY2,along=3), c(1,2), mean)
#then plot as raster

You will then see that **the predicted maps are quite different**, map on left is Method 1, map on right is Method 2. I should also note that: gradient$rLNew has 1387 units (same as fitting X data), and rLs has 365 units (same as new X data). Why does prepareGradient give a new random level object with 1387 units, when my new data only has 365 data points?

Method 1 is that used in the Ovaskainen & Abrego 2020, and the prediction looks like expected relationships with my fixed effects (Latitude and Depth). 
Method 2 on the other hand predicts more fine-scale variation. Is Method 2 showing something to do with the spatial random effects, or is that noise from an error in my syntax?
**In short, is Method 1 or Method 2 most accurate, and most accurately representing the spatial random effects?**

Thanks!

![Prediction syntax approaches](https://github.com/hmsc-r/HMSC/assets/6924440/db0e0ded-2895-4b4b-bf29-f839b6a19420)
